### PR TITLE
Add howto about migrating a package to package spec v2

### PR DESCRIPTION
**Pull request by jsoriano**
Add documentation about how to solve validation issues found with packages following the package spec v2.

Part of https://github.com/elastic/package-spec/issues/399.
Fixes https://github.com/elastic/elastic-package/issues/963 (additional implemented validations should include their own docs in this document).

### CODE REVIEW

**Review by mrodm — APPROVED** [review]
LGTM
And those examples with possible errors will help for sure developers to upgrade the packages from v1 to v2 :+1:

**Comment by mtojek** [comment]
@v1v It looks like the job is still being triggered by beats-ci? 

https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Felastic-package-package-storage-publish/detail/PR-964/2/pipeline

File: docs/howto/update_major_package_spec.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Update between Package Spec major versions
+
+Each Package Spec major version may bring breaking changes or additional
+validations that may make tests to fail for some packages. Find here guidelines
+about how to fix these issues.
+
+Version of the Package Spec used by a package is defined by the `format_version`
+setting in the `manifest.yml` file.
+
+## Upgrading from Package Spec v1 to v2
```
Comment (mtojek) [inline docs/howto/update_major_package_spec.md:10]: nit: I would add a prefix "Troubleshooting" as this is rather information about how to fix an issue.

**Review by mtojek — APPROVED** [review]

**Comment by jsoriano** [comment]
Failure is unrelated, merging.

**Pull request merged**